### PR TITLE
Add mysterious stranger sounds to detective's revolver

### PR DESCRIPTION
**Pull request by JohnGinnane**
Now the detective can try and RP as a cool noir guy

[Changelogs]: Added a proc to handle the different ways in which a weapon is picked up, dropped or equipped. Had to add a minor delay because of the order of some procs and when the item is in the hand or not, etc.

:cl: JohnGinnane
soundadd: Detective's gun now has unique sounds for when he holsters and unholsters it
/:cl:

[why]: A little audio flavour would be nice

### CODE REVIEW

**Comment by ghost** [comment]
We really need more sound effects

**Comment by JohnGinnane** [comment]
image

File: hippiestation/code/modules/projectiles/guns/ballistic/revolver.dm [hunk 1]
```diff
@@ -6,3 +6,36 @@
 
 /obj/item/gun/ballistic/revolver/doublebarrel
 	fire_sound = 'hippiestation/sound/weapons/shotgun_shoot.ogg'
+
+/obj/item/gun/ballistic/revolver/detective
```
Comment (ghost) [inline hippiestation/code/modules/projectiles/guns/ballistic/revolver.dm]: Put all of this on `/obj/item/gun` and put it in a new dm file so we can reuse it

**Review by ghost — DISMISSED** [review]

File: hippiestation/code/modules/projectiles/guns/ballistic/revolver.dm (hunk 1 above)
Comment (JohnGinnane) [reply hippiestation/code/modules/projectiles/guns/ballistic/revolver.dm]: It's already in one of our own versions though?

**Comment by ghost** [comment]
I like the idea but it'd be nice if it only did this in fully bright environments?

I think pulling the gun out in stealth while in maint and it being loud is bad.

**Comment by JohnGinnane** [comment]
Not a bad idea, I didn't consider people using it in a sneak attack. Let me work on that

**Comment by JohnGinnane** [comment]
That's done. It'll only play when the light level is more than 0.4 (out of max 1). To give you some idea of this the character below is standing on 0.403

image

**Comment by ghost** [comment]
great
I'm good with this

**Comment by JohnGinnane** [comment]
turns out this behaviour is already implemented:

image

**Comment by JohnGinnane** [comment]
This is still WIP, will have it fixed later on today

**Comment by JohnGinnane** [comment]
This is ready now I think

**Review by KayeArray — APPROVED** [review]
k

**Comment by KayeArray** [comment]
+1

**Comment by bardofrage** [comment]
+1 o'ne.

**Comment by JohnGinnane** [comment]
thanks bruh just need someone to merge because self merge is 🤸‍♂️ 🚫 🆗

**Pull request merged**